include/hmap/dynamic-hmap.hpp

```diff
@@ -365,6 +365,10 @@ class DynamicHMap {
 		return std::make_pair(boost::make_transform_iterator<AnyCaster<V> >(iter),
 							  inserted);
 	}
+	template<typename A>
+	auto unsafe_insert_or_assign(const detail::KeyBase& kB, A&& a) {
```

Comment:
We do probably want to implement the iterator-hinted version of this as well, because that's the one you'll need to get a linear rebuild from sequential insertions.

Reply:
True.

Reply:
The other thing a hinted version of unsafe_insert_or_assign can do is an efficient merge between existing entries and new entries.  The hint should point at the first node with a higher key than the one you are inserting (or cend() if there are none).

Unlike the normal DynamicHMap::try_emplace and DynamicHMap::insert_or_assign, unsafe_insert_or_assign should implement all four overloads of the original insert_or_assign, not just one or two... both hinted and unhinted and also with both `const Detail::KeyBase&` and `Detail::KeyBase&&`.

Reply:
@johnrgrout and @elfprince13 , what's the verdict here?